src/auth/auth.service.ts

```diff
@@ -0,0 +1,73 @@
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { ApiMessages } from 'src/common/constants';
+import { Env } from 'src/env';
+import axios from 'axios';
+import { GithubService } from 'src/github/github.service';
+import { TRequest } from 'src/common/types/user.types';
+import { RedisService } from '@liaoliaots/nestjs-redis';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private readonly jwtService: JwtService,
+    private readonly githubService: GithubService,
+    private readonly redisService: RedisService,
+  ) {}
+
+  async handleGitHubCallback(code: string) {
+    const clientId = Env.GITHUB.CLIENT_ID;
+    const clientSecret = Env.GITHUB.CLIENT_SECRET;
+
+    try {
+      const tokenResponse = await axios.post(
+        'https://github.com/login/oauth/access_token',
+        {
+          client_id: clientId,
+          client_secret: clientSecret,
+          code,
+        },
+        {
+          headers: {
+            accept: 'application/json',
+          },
+        },
+      );
+
+      const accessToken = tokenResponse.data.access_token;
+      if (!accessToken) {
+        throw new HttpException(ApiMessages.NOT_FOUND, HttpStatus.NOT_FOUND);
+      }
+
+      const userDetails = await this.githubService.getUserDetails(accessToken);
+      await this.githubService.saveUserDetails(userDetails, accessToken);
+      const jwtToken = this.jwtService.sign(
+        {
+          userId: userDetails.id,
+          email: userDetails.email,
+        },
+        {
+          secret: Env.JWT.AUTH_TOKEN,
+          expiresIn: Env.JWT.EXPIRES_IN,
+        },
+      );
+      const client = this.redisService.getClient();
+      await client.set(
+        `jwt:${userDetails.id}`,
+        jwtToken,
+        'EX',
+        Env.REDIS.EXPIRY,
+      );
+      await this.githubService.saveToken(userDetails.id, jwtToken);
+      return jwtToken;
+    } catch (error) {
+      throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+    }
```

Comment:
PullProofAI

There is a potential security vulnerability in the `handleGitHubCallback` method where the GitHub access token is being stored in Redis without proper validation or sanitization.
```typescript
18:   async handleGitHubCallback(code: string) {
...
42:       const userDetails = await this.githubService.getUserDetails(accessToken);
43:       await this.githubService.saveUserDetails(userDetails, accessToken);
...
56:         `jwt:${userDetails.id}`,
57:         jwtToken,
...
61:       await this.githubService.saveToken(userDetails.id, jwtToken);
...
```

#### Suggested fix:
Ensure that sensitive data like access tokens are properly handled and stored securely. Consider encrypting the token before storing it in Redis or using a more secure method for token storage.

This can involve:
1. Encrypting the token before storing it.
2. Implementing proper token validation and expiration mechanisms.
3. Using secure storage solutions for sensitive data.

It's crucial to follow best practices for handling sensitive information to prevent security vulnerabilities.


<!-- This is an auto-generated comment by PullProofAI -->